provider/cmd/pulumi-gen-kubernetes/main.go

```diff
@@ -93,118 +105,71 @@ func main() {
-	if err != nil {
-		panic(err)
-	}
-
-	for groupName, group := range groupsts {
-		groupDir := fmt.Sprintf("%s/%s", outdir, groupName)
-		err = os.MkdirAll(groupDir, 0700)
-		if err != nil {
-			panic(err)
+	templateResources := gen.TemplateResources{}
+	for _, resource := range resources {
+		if resource.Package == "" {
+			continue
 		}
-
-		for versionName, version := range group.Versions {
-			versionDir := fmt.Sprintf("%s/%s", groupDir, versionName)
-			err = os.MkdirAll(versionDir, 0700)
-			if err != nil {
-				panic(err)
-			}
-
-			for kindName, kind := range version.Kinds {
-				err = ioutil.WriteFile(fmt.Sprintf("%s/%s.ts", versionDir, kindName), []byte(kind), 0777)
-				if err != nil {
-					panic(err)
-				}
-			}
-
-			err = ioutil.WriteFile(fmt.Sprintf("%s/%s.ts", versionDir, "index"), []byte(version.Index), 0777)
-			if err != nil {
-				panic(err)
-			}
+		tr := gen.TemplateResource{
+			Name:    resource.Name,
+			Package: resource.Package,
+			Token:   resource.Token,
 		}
-
-		err = ioutil.WriteFile(fmt.Sprintf("%s/%s.ts", groupDir, "index"), []byte(group.Index), 0777)
-		if err != nil {
-			panic(err)
+		for _, property := range resource.Properties {
+			tp := gen.TemplateProperty{
+				ConstValue: property.ConstValue,
+				Name:       property.Name,
+				Package:    property.Package,
+			}
+			tr.Properties = append(tr.Properties, tp)
 		}
+		templateResources.Resources = append(templateResources.Resources, tr)
 	}
+	sort.Slice(templateResources.Resources, func(i, j int) bool {
+		return templateResources.Resources[i].Token < templateResources.Resources[j].Token
+	})
 
-	err = ioutil.WriteFile(fmt.Sprintf("%s/index.ts", outdir), []byte(indexts), 0777)
-	if err != nil {
-		panic(err)
+	overlays := map[string][]byte{
+		"apiextensions/customResource.ts": mustLoadFile(filepath.Join(templateDir, "apiextensions", "customResource.ts")),
+		"helm/v2/helm.ts":                 mustLoadFile(filepath.Join(templateDir, "helm", "v2", "helm.ts")),
+		"helm/v3/helm.ts":                 mustLoadFile(filepath.Join(templateDir, "helm", "v2", "helm.ts")), // v3 support is currently identical to v2
+		"yaml/yaml.ts":                    mustRenderTemplate(filepath.Join(templateDir, "yaml", "yaml.tmpl"), templateResources),
 	}
-
-	err = ioutil.WriteFile(fmt.Sprintf("%s/package.json", outdir), []byte(packagejson), 0777)
+	files, err := nodejsgen.GeneratePackage("pulumigen", pkg, overlays)
 	if err != nil {
 		panic(err)
 	}
 
-	err = CopyFile(
-		filepath.Join(templateDir, "CustomResource.ts"), filepath.Join(outdir, "apiextensions", "CustomResource.ts"))
-	if err != nil {
-		panic(err)
-	}
-
-	err = CopyFile(filepath.Join(templateDir, "README.md"), filepath.Join(outdir, "README.md"))
-	if err != nil {
-		panic(err)
-	}
+	// Internal files that don't need to be exported
+	files["path.ts"] = mustLoadFile(filepath.Join(templateDir, "path.ts"))
+	files["tests/path.ts"] = mustLoadFile(filepath.Join(templateDir, "tests", "path.ts"))
```

Comment:
@pgavlin These two files are not part of the user-facing SDK (i.e. I don't want them in the top level `index.ts`. This has the side-effect of not including them in the `package.json` list of files. Is this ok, or do you have another suggestion?

Reply:
cc @lukehoban

Reply:
I think that's okay.  I think our JS doc generator relies upon public surface area being in package.json, but sounds like this is not public surface area.  I don't know of anything else that relies on the files list in package.json